src/gui/Layout.cpp

```diff
@@ -137,153 +140,131 @@ Rectangle Layout::getVisibleRect()
 	                 gtk_adjustment_get_page_size(scrollHandling->getVertical()));
 }
 
-/**
- * Returns the height of the entire Layout
- */
-double Layout::getLayoutHeight()
+int addif(int lhs, int rhs, bool b)
 {
-	XOJ_CHECK_TYPE(Layout);
-
-	return layoutHeight;
+	if (b)
+	{
+		return lhs + rhs;
+	}
+	return lhs;
 }
 
-/**
- * Returns the width of the entire Layout
- */
-double Layout::getLayoutWidth()
+
+void Layout::recalculate()
 {
 	XOJ_CHECK_TYPE(Layout);
 
-	return layoutWidth;
-}
+	auto* settings = view->getControl()->getSettings();
+	auto len = view->viewPagesLen;
+	mapper.configureFromSettings(len, settings);
+	auto colCount = mapper.getColumns();
+	auto rowCount = mapper.getRows();
```

Comment:
Use auto only with pointers on initialization. Otherwise we have a hard time to:
a) check your code
b) maintaining said code as finding out which type a variable is will be a tedious task.